src/main/java/me/zeroeightsix/kami/util/astaralg/Astarpathfinder.java

```diff
@@ -0,0 +1,650 @@
+        }
+        enabled = false;
+        path = null;
+    }
+
+
+    public static void onreceive(PacketEvent.Receive event) {
+        if (!enabled) return;
+        if (event.getPacket() instanceof SPacketEntityMetadata) {
+            SPacketEntityMetadata meta = (SPacketEntityMetadata) event.getPacket();
+            if (meta.getEntityId() == mc.player.getEntityId()) {
+                for (EntityDataManager.DataEntry d : meta.getDataManagerEntries()) {
+                    if (d.getKey().getId() == 11 && ((float) d.getValue()) < mc.player.getAbsorptionAmount())
+                        checkstuckInBlock();
+
+                }
+            }
+        } else if (event.getPacket() instanceof SPacketUpdateHealth) {
+            SPacketUpdateHealth packet = (SPacketUpdateHealth) event.getPacket();
+            if (packet.getHealth() < mc.player.getHealth()) {
+                checkstuckInBlock();
+            }
+        } else if (event.getPacket() instanceof SPacketPlayerPosLook) {
+            SPacketPlayerPosLook packet = (SPacketPlayerPosLook) event.getPacket();
+            if ((Math.abs(packet.x - laspacketx) < 0.05 && Math.abs(packet.z - laspacketz) < 0.05)) {
+                rubberbandcounter++;
+            } else {
+                rubberbandcounter = 0;
+            }
+            laspacketx = packet.x;
+            laspacketz = packet.z;
+        }
+    }
+
+    private static void checkstuckInBlock() {
+        boolean condition = enabled &&
+                ((state == State.PATHFIND || (state == State.LAND_SAVE && !landed)) &&
+                        !mc.player.onGround &&
+                        !mc.player.isElytraFlying());
+
+        //for testing!!! can be removed after
+        MessageSendHelper.sendChatMessage("Checking if blockstuck:" + (condition ? (TextFormatting.RED + " Stuck") : (TextFormatting.GREEN + " Only Damage")));
+        if (condition) {
+            blockstuck = true;
+            mc.timer.tickLength = 50.0f;
+            mc.player.motionZ = 0;
+            mc.player.motionX = 0;
+            mc.player.motionY = -0.4;
+            mc.player.capabilities.isFlying = false;
+            mc.player.capabilities.setFlySpeed(0.05f);
+        }
+    }
+
+    public static void update(PlayerTravelEvent event) {
+        currentEvent = event;
+        try {
+            switch (state) {
+                case START_FLYING:
+                    startFlying();
+                    break;
+                case PATHFIND:
+                    pathfind();
+                    break;
+                case GOTO_TRACK:
+                    gotoTrack();
+                    break;
+                case LAND_SAVE:
+                    landSave();
+                    break;
+            }
+        } catch (Exception e) {
+            MessageSendHelper.sendChatMessage("Error occurred send logs in discord ");
+            //i mean try catch barely uses performance
```

Comment:
ngl i'm really wondering how this would ever throw an exception

Reply:
idk i can remove it if you want